examples/operator.rs

```diff
@@ -0,0 +1,100 @@
+extern crate timely;
+extern crate timely_communication;
+
+use std::collections::HashMap;
+
+use timely_communication::Configuration;
+use timely::dataflow::operators::{ToStream, Input, Inspect, Operator, FrontierNotificator};
+use timely::dataflow::channels::pact::Pipeline;
+use timely::progress::timestamp::RootTimestamp;
+use timely::dataflow::Scope;
+
+fn main() {
+    timely::execute(Configuration::Thread, |root| {
+        root.scoped(|scope| {
+            (0u64..10).to_stream(scope)
+                .unary_frontier(Pipeline, "example", |mut builder| {
+                    let mut cap = Some(builder.get_cap(RootTimestamp::new(12)));
+                    move |(input, _frontier), output| {
+                        cap = None;
+                        while let Some((time, data)) = input.next() {
+                            output.session(&time).give_content(data);
+                        }
+                    }
+                }).inspect(|x| println!("{:?}", x));
+        });
+    }).unwrap();
+
+    timely::execute(Configuration::Thread, |root| {
+        root.scoped(|scope| {
+            (0u64..10).to_stream(scope)
+                .unary(Pipeline, "example", |mut builder| {
```

Comment:
Read things. Correct me if I'm wrong, but it looks like the trait exists because these two have similar methods, but no code uses the trait for polymorphism. So, we could skip the trait and put the methods in the `impl` for each type, if we wanted to skip the `use` statements.